http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/NinetyNineBottlesOfBeerResource.java

```diff
@@ -0,0 +1,83 @@
+    public static final String ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_VAL = "true";
+    public static final String FIRST_BOTTLE_RESPONSE = "1 bottle of beer on the wall, 1 bottle of beer. Take one down, pass it around, no more beer on the wall!";
+    public static final String SECOND_BOTTLE_RESPONSE = "2 bottles of beer on the wall, 2 bottles of beer. Take one down, pass it around, 1 bottle of beer on the wall.";
+    public static final String OTHER_BOTTLES_RESPONSE = "%d bottles of beer on the wall, %d bottles of beer. Take one down, pass it around, %d bottles of beer on the wall.";
+    public static final String PATH = "/99-bottles-of-beer/bottle";
+    public static final String APPLICATION_YAML = "application/yaml";
+    private static final String EXCEPTION_MESSAGE = "%s: expected path parameter value '%s' differ from '%s'.";
+    private static final String IMAGE_PNG = "image/png";
+    private static final String IMAGE_JPEG = "image/jpeg";
+    private static final String TEXT_CSS = "text/css";
+    private static final String TEXT_XML = "text/xml";
+
+    @Produces(APPLICATION_JSON)
+    @GET
+    @Path("/{bottle-number:1}")
+    public Uni<String> getFirstBottle(@PathParam("bottle-number") Integer bottleNumber) {
+        if (bottleNumber != null && bottleNumber.equals(1)) {
+            return Uni.createFrom().item(FIRST_BOTTLE_RESPONSE);
+        }
+        throw new IllegalStateException(String.format(EXCEPTION_MESSAGE, "getFirstBottle", 1, bottleNumber));
+    }
+
+    @Produces(APPLICATION_JSON)
+    @GET
+    @Path("/{2-nd-bottle-number:[^0-1,3-9]}")
+    public Uni<String> getSecondBottle(@PathParam("2-nd-bottle-number") Integer bottleNumber) {
+        if (bottleNumber != null && bottleNumber.equals(2)) {
+            return Uni.createFrom().item(SECOND_BOTTLE_RESPONSE);
+        }
+        throw new IllegalStateException(String.format(EXCEPTION_MESSAGE, "getSecondBottle", 2, bottleNumber));
+    }
+
+    @Produces(APPLICATION_JSON)
+    @GET
+    @Path("/{_bottle.number:[3-9]|[1-9][0-9]}")
+    public Uni<String> getOtherBottles(@PathParam("_bottle.number") Integer bottleNumber) {
+        if (bottleNumber != null && bottleNumber >= 3 && bottleNumber <= 99) {
+            return Uni.createFrom().item(String.format(OTHER_BOTTLES_RESPONSE, bottleNumber, bottleNumber, bottleNumber - 1));
+        }
+        throw new IllegalStateException(String.format(EXCEPTION_MESSAGE, "getOtherBottles", "3-99", bottleNumber));
+    }
+
+    /**
+     * Path parameter is intentionally not used as that's allowed by JAX-RS.
+     */
+    @Produces({ APPLICATION_JSON, APPLICATION_XML, TEXT_HTML, TEXT_PLAIN, MediaType.APPLICATION_OCTET_STREAM,
+            MediaType.MULTIPART_FORM_DATA, IMAGE_PNG, IMAGE_JPEG, APPLICATION_YAML, TEXT_CSS, TEXT_XML })
+    @GET
+    @Path("/{bottle-number:[1-9][0-9]{2,}}")
+    public Response getBeerBox() {
```

Comment:
This endpoint is used for tests that are not concerned with `@Path` parameters parsing. By keeping it in the same resource class, you are limiting the Quarkus versions, which the corresponding tests can be ran against, even though it is not necessary.
I think it would be better to extract this endpoint to a separate resource, use a simple `@Path`, and then get rid of the `@EnabledIfSystemProperty` in the two affected tests:
- `HttpAdvancedReactiveIT#messageBodyWriter`
- `HttpAdvancedReactiveIT#responseContentType`

Reply:
Sure, you're right